### DISCUSSION TITLE
X axis annotations not working with custom xaxis categories

### DISCUSSION THREAD

**Discussion by DhrunitSixberries** [apexcharts/apexcharts.js#3571] [General]
I am trying to show an xaxis annotation with custom x axis categories

my x axis categories are 
`
  xaxis: {
            type: "numeric",
            categories: [6, 8, 13, 18]
            }
`

and my component is as below it is a line chart component

import "./styles.css";
import Chart from "react-apexcharts";
import { TextField, MenuItem } from "@material-ui/core";

export default function App() {
  return (
    <div className="App" style={{ position: "relative", maxWidth: "500px" }}>
      <Chart
        options={{
          chart: {
            height: 350,
            type: "line",
            zoom: {
              enabled: false
            }
          },
          dataLabels: {
            enabled: false
          },
          markers: {
            size: 5,
            colors: undefined,
            strokeColors: "#fff",
            strokeWidth: 1,
            strokeOpacity: 0.5,
            strokeDashArray: 0,
            fillOpacity: 1,
            discrete: [],
            shape: "circle",
            radius: 6,
            offsetX: 0,
            offsetY: 0,
            onClick: undefined,
            onDblClick: undefined,
            showNullDataPoints: true,
            hover: {
              size: undefined,
              sizeOffset: 3
            }
          },
          stroke: {
            curve: "straight"
          },
          title: {
            text: "Product Trends by Month",
            align: "left"
          },
          grid: {
            row: {
              colors: ["#f3f3f3", "transparent"], // takes an array which will be repeated on columns
              opacity: 0.5
            }
          },
          annotations: {
            xaxis: [
              {
                x: 9,
                strokeDashArray: 5,
                borderColor: "red",
                borderWidth: 2,
                label: {
                  borderColor: "red",
                  style: {
                    color: "#fff",
                    background: "red"
                  }
                }
              }
            ]
          },
          xaxis: {
            type: "category",
            tickAmount: "dataPoints",
            categories: [6, 8, 13, 18]
          }
        }}
        series={[
          {
            name: "Desktops",
            data: [10, 8, 10, 10]
          }
        ]}
        type="line"
        width="500"
      />
      {/* <div style={{ position: "absolute", left: "50%" }}>12% incidence</div> */}
    </div>
  );
}

`
annotations: {
            xaxis: [
              {
                x: 9,
                strokeDashArray: 5,
                borderColor: "red",
                borderWidth: 2,
                label: {
                  borderColor: "red",
                  style: {
                    color: "#fff",
                    background: "red"
                  }
                }
              }
            ]
          },
`

As you can see above I am trying to have the annotation at the point **9** on the x axis but I am not getting the annotation at the right place
see output below

![image](https://user-images.githubusercontent.com/92073905/209548420-91edd7bd-e18b-4c87-9259-4bb731389a55.png)

[here](https://codesandbox.io/s/line-chart-with-dots-comq9z?file=/src/App.js) is the code sandbox link 

(Sorry I am not good at formatting code)

**Comment by Yalcingezsat** [comment]
It looks like you are trying to show the annotation at x position 9, but the x axis categories only go up to 18. You should change the x position of the annotation to one of the values in the categories array, such as 6, 8, 13, or 18.

Additionally, it appears that you are using the type: "category" property for the x axis, but you are also specifying xaxis: { type: "numeric" } earlier in the options object. You should remove the type: "numeric" property to avoid any conflicts.

Here is the modified version of your code:


`options={{ chart: { height: 350, type: "line", zoom: { enabled: false } }, dataLabels: { enabled: false }, markers: { size: 5, colors: undefined, strokeColors: "#fff", strokeWidth: 1, strokeOpacity: 0.5, strokeDashArray: 0, fillOpacity: 1, discrete: [], shape: "circle", radius: 6, offsetX: 0, offsetY: 0, onClick: undefined, onDblClick: undefined, showNullDataPoints: true, hover: { size: undefined, sizeOffset: 3 } }, stroke: { curve: "straight" }, title: { text: "Product Trends by Month", align: "left" }, grid: { row: { colors: ["#f3f3f3", "transparent"], // takes an array which will be repeated on columns opacity: 0.5 } }, annotations: { xaxis: [ { x: 8, // changed x position to one of the x axis categories strokeDashArray: 5, borderColor: "red", borderWidth: 2, label: { borderColor: "red", style: { color: "#fff", background: "red" } } } ] }, xaxis: { type: "category", // removed "numeric" type property tickAmount: "dataPoints", categories: [6, 8, 13, 18] } }}`


Your question was also shared on AnswerDeveloper.

Answer Link:  https://www.answerdeveloper.com/questions-answers/other-programming-language/x-axis-annotations-not-working-with-custom-xaxis-categories

**Comment by DhrunitSixberries** [comment]
> It looks like you are trying to show the annotation at x position 9, but the x axis categories only go up to 18. You should change the x position of the annotation to one of the values in the categories array, such as 6, 8, 13, or 18.
> 
> Additionally, it appears that you are using the type: "category" property for the x axis, but you are also specifying xaxis: { type: "numeric" } earlier in the options object. You should remove the type: "numeric" property to avoid any conflicts.
> 
> Here is the modified version of your code:
> 
> `options={{ chart: { height: 350, type: "line", zoom: { enabled: false } }, dataLabels: { enabled: false }, markers: { size: 5, colors: undefined, strokeColors: "#fff", strokeWidth: 1, strokeOpacity: 0.5, strokeDashArray: 0, fillOpacity: 1, discrete: [], shape: "circle", radius: 6, offsetX: 0, offsetY: 0, onClick: undefined, onDblClick: undefined, showNullDataPoints: true, hover: { size: undefined, sizeOffset: 3 } }, stroke: { curve: "straight" }, title: { text: "Product Trends by Month", align: "left" }, grid: { row: { colors: ["#f3f3f3", "transparent"], // takes an array which will be repeated on columns opacity: 0.5 } }, annotations: { xaxis: [ { x: 8, // changed x position to one of the x axis categories strokeDashArray: 5, borderColor: "red", borderWidth: 2, label: { borderColor: "red", style: { color: "#fff", background: "red" } } } ] }, xaxis: { type: "category", // removed "numeric" type property tickAmount: "dataPoints", categories: [6, 8, 13, 18] } }}`
> 
> Your question was also shared on AnswerDeveloper.
> 
> Answer Link: https://www.answerdeveloper.com/questions-answers/other-programming-language/x-axis-annotations-not-working-with-custom-xaxis-categories

My use case is having annotation to show on the x axis at point 9 and not on point 8

**Comment by Yalcingezsat** [comment]
@DhrunitSixberries  I submitted your problem and got a reply like this from AnswerDeveloper



If you want to show the annotation at point 9 on the x-axis, you need to set the x value of the annotation to 9. Currently, the x value is set to 9 but the x-axis is using categories with values 6, 8, 13, and 18.

To fix this, you can either change the x value of the annotation to one of the categories or you can change the x-axis to use numeric values instead of categories.

To change the x value of the annotation to one of the categories, you can set it to the index of the category in the array. For example, to show the annotation at the third category (13), you can set x to 2:

annotations: {
xaxis: [
{
x: 2,
strokeDashArray: 5,
borderColor: "red",
borderWidth: 2,
label: {
borderColor: "red",
style: {
color: "#fff",
background: "red"
}
}
}
]
},

To change the x-axis to use numeric values, you can remove the type: "category" option and set the type: "numeric":

xaxis: {
type: "numeric",
tickAmount: "dataPoints",
categories: [6, 8, 13, 18]
},

Then you can set the x value of the annotation to the numeric value you want to show it at:

```
annotations: {
xaxis: [
{
x: 9,
strokeDashArray: 5,
borderColor: "red",
borderWidth: 2,
label: {
borderColor: "red",
style: {
color: "#fff",
background: "red"
}
}
}
]
},

```


https://www.answerdeveloper.com/questions-answers/main/comment/c3a72619-b2ad-4d6c-a7fc-a937aa4eb5eb?postId=63a993da72bd7000129b7726

**Comment by DhrunitSixberries** [comment]
@Yalcingezsat Thank you for your reply I tried your approach this is what i have as an output
![image](https://user-images.githubusercontent.com/92073905/209552022-8be923ae-c705-4fe8-8d26-94ff29cb55d0.png)
I want [6, 8, 13, 18] in the x axis and the annotation on the point 9 right now I have different values for x axis
